lib/paramedic.js

```diff
@@ -190,48 +192,160 @@ ParamedicRunner.prototype.writeMedicConnectionUrl = function(url) {
+    .then(function(command) {
+        self.setPermissions();
         logger.normal('cordova-paramedic: running command ' + command);
 
-        return execPromise(command)
-        .then(self.runSauceTests.bind(self), function(output) {
-            // this trace is automatically available in verbose mode
-            // so we check for this flag to not trace twice
-            if (!self.config.verbose) {
-                logger.normal(output);
-            }
-            logger.normal('cordova-paramedic: unable to build project; command log is available above');
-            throw new Error('Command "' + command + '" failed.');
+        return execPromise(command);
+    })
+    .then(function() {
+        // skip tests if it was just build
+        if (self.shouldWaitForTestResult()) {
+            return Q.promise(function(resolve, reject) {
+                // reject if timed out
+                self.waitForConnection().catch(reject);
+                // resolve if got results
+                self.waitForTests().then(resolve);
+            });
+        }
+    }, function(output) {
+        // this trace is automatically available in verbose mode
+        // so we check for this flag to not trace twice
+        if (!self.config.verbose) {
+            logger.normal(output);
+        }
+        logger.normal('cordova-paramedic: unable to run tests; command log is available above');
+        throw new Error('Unable to run tests.');
+    });
+};
+
+ParamedicRunner.prototype.getAppiumDeviceName = function () {
+    if (!this.targetObj) {
+        return '';
+    }
+    if (this.config.getPlatformId() === 'ios') {
+        return this.targetObj.target.replace('-', ' ');
+    }
+    return this.targetObj.target;
+};
+
+ParamedicRunner.prototype.runAppiumTests = function () {
+    var platform = this.config.getPlatformId();
+    if (platform !== 'android' && platform !== 'ios') {
+        logger.warn('Unsupported platform for Appium test run: ' + platform);
+        return Q(true);
+    }
+    var self = this;
+
+    logger.normal('Running Appium tests locally.');
+
+    var options = {
+        platform: self.config.getPlatformId(),
+        appPath: self.tempFolder.name,
+        appiumDeviceName: self.getAppiumDeviceName(),
+        appiumPlatformVersion: null,
+        device: false,
+        screenshotPath: path.join(process.cwd(), 'appium_screenshots'),
+        output: self.config.getOutputDir(),
+        verbose: self.config.isVerbose()
+    };
+
+    var appiumRunner = new AppiumRunner(options);
+    if (appiumRunner.options.testPaths && appiumRunner.options.testPaths.length === 0) {
+        logger.warn('Couldn\'t find Appium tests, skipping...');
+        return Q(true);
+    }
+    return Q()
+    .then(function () {
+        return appiumRunner.prepareApp();
+    })
+    .then(function () {
+        return appiumRunner.runTests();
+    });
+
+};
+
+ParamedicRunner.prototype.runAppiumTestsOnSauce = function () {
```

Comment:
This method looks very similar to `runAppiumTests` above. Consider deduplicating these two methods to the single one that accepts boolean `runOnSauce` as a param?